mujtaba_charm/utils/sample.py

```diff
@@ -0,0 +1,30 @@
+"""
+This module contains the hello function.
+"""
+
+def hello(name="world"):
+    """
+    Returns a string greeting the name entered as parameter. If no name
+    provided, it greets the world.
+
+    Args:
+        name (str, optional): The name that is greeted by the function. Default parameter is "world".
+
+    Returns:
+        str: A string that greets the user.
+
+    Raises:
+        TypeError: If the argument is not of string type.
+
+    Example:
+        >>> hello("john")
+        'hello john!'
+
+        >>> hello()
+        'hello world!'
+    """
+
+    if not isinstance(name, str):
+        raise TypeError("name should be of string type")
```

Comment:
also let the user know what data type they passed

Reply:
Done by using type(name).__name__ to avoid <class int> and just display int